### DISCUSSION TITLE
GitHub Copilot 'Code Completion' is not working in my Visual Studio IDE

### DISCUSSION THREAD

**Discussion by fsd8fa0sd** [community/community#128345] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

While other Copilot features seem to function normally in Visual Studio professional v17.10, like code suggestions within the chat using 'alt + /', but 'code completion' has never worked by using 'Tab' or 'Enter' key as it did in VS code or documented in https://learn.microsoft.com/en-us/visualstudio/ide/visual-studio-github-copilot-extension?view=vs-2022.   I've tried to adjust settings in Tools -> Options -> Text Editor and even disabled IntelliCode (another code completion tool) in case of any conflicts, but the issue persists. Is it an issue with my github copilot subscription, or any other suggestion to fix the issue? Thanks

**Comment by ayim** [comment]
Hi there,

1) The IntelliCode line completions should be enabled, Copilot relies on this component to deliver grey text
2) What languages are you writing code in? We had a known issue with the Tab key in Markdown but this shouldn't apply to other languages

If you continue to experience issues, please write me at aayim @ microsoft.com with your availability over the next week / time zone so that we can screenshare and see what's going on.

--
Aaron Yim
Product Manager for GitHub Copilot in Visual Studio

**Reply by ayim** [reply]
@fsd8fa0sd ping on this here! We'd love to get this working.

First steps, please **enable** IntelliCode whole line autocomplete. Copilot relies on this component to deliver grey-text.

--
Aaron Yim
Product Manager for GitHub Copilot in Visual Studio
aayim@microsoft.com

**Reply by fsd8fa0sd** [reply]
![image](https://github.com/community/community/assets/55037387/c8c2efb2-e385-4169-9bdf-c75b68fc0059)
I enabled  'IntelliCode whole line autocomplete'. Code completion didn't work for both C# and python. But other chat functions like ALT + / are working as expected.

**Reply by ayim** [reply]
Gotcha, could you confirm that both of these are enabled?

![image](https://github.com/user-attachments/assets/bba111cb-f998-4141-8c8c-60acb08a461b)
![image](https://github.com/user-attachments/assets/c81abb66-2ddc-4510-ae1a-ab5d22f28134)

**Reply by fsd8fa0sd** [reply]
Yes, both settings are enabled
![image](https://github.com/community/community/assets/55037387/0ca496ac-b7d6-4de3-9539-0c46a92cdfad)

![image](https://github.com/community/community/assets/55037387/512f5f64-d7d2-409e-9f6e-43973df6611e)

and I think that it is related to some firewall rules because it appears to work without the firewall now. Could you kindly advise what are the urls for copilot to be whitelisted?

**Reply by ayim** [reply, marked as answer]
Here's the full list of URLs for the firewall:
https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-firewall-settings-for-github-copilot

**Reply by zeyad-abdelnaby** [reply]
@ayim i have same problem and enabled all settings that can affect 
and it was working until the day before last night

**Reply by Baer4900** [reply]
Same here after Updating Visual Insider to latest Version

**Reply by WCRev** [reply]
@fsd8fa0sd did you ever fix it? I am having the same issue with 2022 Professional, but it's working with 2022 Enterprise and 20226 Professional

**Reply by Baer4900** [reply]
I have resetted the Shortcut Settings (Environment -> Keyboard -> reset) Since then it works again.
